lib/watir-webdriver/locators/element_locator.rb

```diff
@@ -233,6 +242,21 @@ def delete_regexps_from(selector)
       rx_selector
     end
 
+    def regexp_selector_to_predicates(key, re)
+      return [] if re.casefold?
+
+      # convert the head and trailing runs of literals to `contains` expressions
+      match = re.source.match(/^([^\[\]\\^$.|?*+()]*)[^|]*?([^\[\]\\^$.|?*+()]*)$/)
```

Comment:
I believe it makes sense to:

1. Extract this regexp to consant/method.
2. Split the regexp to multiple lines with comments for each meaningful part.
3. Use `\A\z` instead of `^$`